### PR TITLE
[GENX]: Add test file for GENX dialect operations

### PR DESCRIPTION
**Pull request by etiotto**
Added mlir/test/Dialect/LLVMIR/genx.mlir

### CODE REVIEW

**Comment by etiotto** [comment]
Depends on https://github.com/pengtu/llvm-project/pull/1

File: mlir/lib/Conversion/GPUToGENX/LowerGpuOpsToGENXOps.cpp [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+      RewritePatternSet patterns(ctx);
+      populateGpuRewritePatterns(patterns);
+      if (failed(applyPatternsAndFoldGreedily(m, std::move(patterns))))
+        return signalPassFailure();
+    }
+
+    LLVMTypeConverter converter(ctx, options);
+    populateGpuMemorySpaceAttributeConversions(
+        converter, [](gpu::AddressSpace space) {
+          switch (space) {
```
Comment (pengtu) [inline mlir/lib/Conversion/GPUToGENX/LowerGpuOpsToGENXOps.cpp]: The address space enum is already defined in GENXOps.td.  Please use them:          switch (space) {
          case gpu::AddressSpace::Global:
            return GENX::GENXDialect::kGlobalMemoryAddressSpace; 
          case gpu::AddressSpace::Workgroup:
            return GENX::GENXDialect::kSharedMemoryAddressSpace;
          case gpu::AddressSpace::Private:
            return GENX::GENXDialect::kPrivateMemoryAddressSpace;

File: mlir/lib/Conversion/GPUToGENX/LowerGpuOpsToGENXOps.cpp [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+  patterns.add<ScalarizeVectorOpLowering<OpTy>>(converter);
+  patterns.add<OpToFuncCallLowering<OpTy>>(converter, f32Func, f64Func);
+}
+
+void mlir::populateGpuToGENXConversionPatterns(LLVMTypeConverter &converter,
+                                               RewritePatternSet &patterns) {
+  populateWithGenerated(patterns);
+  // TODO: Add patterns
+  //  patterns
+  //    .add<GPUIndexIntrinsicOpLowering<gpu::ThreadIdOp, GENX::ThreadIdXOp,
```
Comment (pengtu) [inline mlir/lib/Conversion/GPUToGENX/LowerGpuOpsToGENXOps.cpp]: The GPUIndexIntrinsicOpLowering patterns should already work with the current GENXOps.td.  Have you tried to enable them?

**Review by pengtu — CHANGES_REQUESTED** [review]

File: mlir/lib/Conversion/GPUToGENX/LowerGpuOpsToGENXOps.cpp (hunk 2 above)
Comment (etiotto) [reply mlir/lib/Conversion/GPUToGENX/LowerGpuOpsToGENXOps.cpp]: Yes they work but they are enabled in the next PR (https://github.com/pengtu/llvm-project/pull/3) witch also adds a lit test.

**Review by whitneywhtsang — APPROVED** [review]
6f506ee36f4d2003a0f82ab1bdba7a28bbed4922 LGTM.

File: mlir/lib/Conversion/GPUToGENX/LowerGpuOpsToGENXOps.cpp (hunk 1 above)
Comment (pengtu) [reply mlir/lib/Conversion/GPUToGENX/LowerGpuOpsToGENXOps.cpp]: Are you going to make the change in this PR?

**Review by pengtu — APPROVED** [review]
Looks great!

**Pull request merged**